### PR TITLE
[CARBONDATA-1787] Validation for table properties in Create Table Command

### PR DESCRIPTION
**Pull request by geetikagupta16**
In this PR, I have added validation for the options in TBLPROPERTIES for CREATE TABLE command. So whenever an invalid option comes, it will throw an error.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? No
 
 - [ ] Any backward compatibility impacted? No
 
 - [ ] Document update required? No

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1626/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2017/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2018/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1627/

**Comment by anubhav100** [comment]
retest his please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1628/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2019/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1630/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1631/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2022/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1637/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2027/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2052/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1662/

**Comment by geetikagupta16** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2056/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1669/

**Comment by geetikagupta16** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2058/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2059/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1671/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/398/

**Comment by geetikagupta16** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/407/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1679/

**Comment by jackylk** [comment]
Please describe this PR, what validation is added?

**Comment by geetikagupta16** [comment]
@jackylk  In this PR, I have added validation for the options in **TBLPROPERTIES** for **CREATE TABLE** command. So whenever an invalid option comes, it will throw an error.

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala [hunk 1]
```diff
@@ -232,6 +233,30 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf, parser: CarbonSpark2SqlParser)
...
   }
 
+  private def validateTableProperties(properties: Map[String, String]): Map[String, String] = {
+    var isSupported = true
+    val invalidOptions = StringBuilder.newBuilder
+    val tableProperties = Seq("DICTIONARY_INCLUDE", "DICTIONARY_EXCLUDE", "NO_INVERTED_INDEX",
+      "SORT_COLUMNS", "TABLE_BLOCKSIZE", "STREAMING", "SORT_SCOPE", "COMMENT", "PARTITION_TYPE",
+      "NUM_PARTITIONS", "RANGE_INFO", "LIST_INFO", "BUCKETNUMBER", "BUCKETCOLUMNS", "TABLENAME")
+    val tblProperties: Map[String, String] = properties.filter { property =>
+      if (!(tableProperties.exists(prop => prop.equalsIgnoreCase(property._1))
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: we can just iterate over the map and check if any of the properties are invalid. No need to create new tblProperties.
Instead of filter, use collect to get invalid properties and check if any property is returned.

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala [hunk 2]
```diff
@@ -190,7 +191,7 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf, parser: CarbonSpark2SqlParser)
     }
 
     val tableProperties = mutable.Map[String, String]()
-    properties.foreach{property => tableProperties.put(property._1, property._2)}
+    validatedProperties.foreach{property => tableProperties.put(property._1, property._2)}
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: i think we can remove this code. After this we will extract the values so no point in creating mutable.Map. Can you please confirm this?

File: integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/V3offheapvectorTestCase.scala [hunk 3]
```diff
@@ -35,7 +35,7 @@ class V3offheapvectorTestCase extends QueryTest with BeforeAndAfterAll {
   //Check query reponse for select * query with no filters
   test("V3_01_Query_01_033", Include) {
      dropTable("3lakh_uniqdata")
-     sql(s"""CREATE TABLE 3lakh_uniqdata (CUST_ID int,CUST_NAME String,ACTIVE_EMUI_VERSION string, DOB timestamp, DOJ timestamp, BIGINT_COLUMN1 bigint,BIGINT_COLUMN2 bigint,DECIMAL_COLUMN1 decimal(30,10), DECIMAL_COLUMN2 decimal(36,10),Double_COLUMN1 double, Double_COLUMN2 double,INTEGER_COLUMN1 int) STORED BY 'carbondata' TBLPROPERTIES('table_blocksize'='128','include_dictionary'='BIGINT_COLUMN1,BIGINT_COLUMN2,DECIMAL_COLUMN1,DECIMAL_COLUMN2,Double_COLUMN1,Double_COLUMN2,INTEGER_COLUMN1,CUST_ID')""").collect
+     sql(s"""CREATE TABLE 3lakh_uniqdata (CUST_ID int,CUST_NAME String,ACTIVE_EMUI_VERSION string, DOB timestamp, DOJ timestamp, BIGINT_COLUMN1 bigint,BIGINT_COLUMN2 bigint,DECIMAL_COLUMN1 decimal(30,10), DECIMAL_COLUMN2 decimal(36,10),Double_COLUMN1 double, Double_COLUMN2 double,INTEGER_COLUMN1 int) STORED BY 'carbondata' TBLPROPERTIES('table_blocksize'='128','dictionary_include'='BIGINT_COLUMN1,BIGINT_COLUMN2,DECIMAL_COLUMN1,DECIMAL_COLUMN2,Double_COLUMN1,Double_COLUMN2,INTEGER_COLUMN1,CUST_ID')""").collect
```
Comment (kunal642) [inline integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/V3offheapvectorTestCase.scala]: revert the unnecessary changes

Comment (geetikagupta16) [reply integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/V3offheapvectorTestCase.scala]: In this I have changed the TBLPROPERTIES option dictionary_include

Comment (kunal642) [reply integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/V3offheapvectorTestCase.scala]: okay

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala (hunk 2 above)
Comment (geetikagupta16) [reply integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: Removing the mutable Map will lead to changes in the parameters of other methods. Should I work on that?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/618/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1846/

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala (hunk 2 above)
Comment (geetikagupta16) [reply integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: Can't remove the Mutable Map as it is being updated in prepareTableModel method

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2215/

**Comment by kunal642** [comment]
We will not validate the create tbl properties as the user can define his own properties as well.
Please close this
